src/tokenizer.jl

```diff
@@ -72,7 +149,7 @@ If the previous token is BOS, leading spaces are removed.
 llama2.c correspondence: decode (l. 418)
 """
```

Comment:
Maybe add the info to the docstring that all token indices are 1-based, which is different from the C implementation. Also add this to the encode function.